### DISCUSSION TITLE
Nuxt3 - "Normal" routes file

### DISCUSSION THREAD

**Discussion by nivv** [nuxt/nuxt#17072] [Questions]
I don't really like the pages directory, since it doesn't work well with how our backend is setup. In nuxt2 we have a normal vue-router file to declare the routes and components.

Is there an equivalent for nuxt3?

Implementation for nuxt2:

```js
import Vue from 'vue'
import VueRouter from 'vue-router'
import routes from './routes/routes'

Vue.use(VueRouter)

export function createRouter () {
    return new VueRouter({
        mode: 'history',
        routes: routes
    })
}
```

**Comment by fusigabs** [comment]
I also prefer using a separate routes file rather than depend on the directory structure. The [Nuxt Router](https://github.com/nuxt-community/router-module) module can be used for this, but it is not yet ready for Nuxt3 as far as I know. I can't wait for it to be compatible :)

**Comment by nathanchase** [comment]
You can manually add routes to your `nuxt.config` like so:
```js
hooks: {
   'pages:extend'(pages) {
      pages.push(
        {
          name: 'CustomPage',
          path: '/custompage',
          file: path.join(__dirname, './path_to_your/custompage.vue'),
        },        
      );
   },
},
```
